optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -554,4 +561,171 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     private ConstraintCollectors() {
     }
 
+    // ************************************************************************
+    // toCollection
+    // ************************************************************************
+
+    /**
+     * Creates constraint collector that will return matched facts as a collection.
+     *
+     * @param collectionFunction never null, provides new instance of the target collection with given initial capacity
+     * @param <A> type of the matched fact, matching the type of elements in the resulting collection
+     * @param <Result> type of the target collection
+     * @return never null
+     */
+    public static <A, Result extends Collection<A>> UniConstraintCollector<A, ?, Result> toCollection(
+            IntFunction<Result> collectionFunction) {
+        return toCollection(Function.identity(), collectionFunction);
+    }
+
+    public static <A> UniConstraintCollector<A, ?, Set<A>> toSet() {
+        return toSet(Function.identity());
+    }
+
+    public static <A> UniConstraintCollector<A, ?, List<A>> toList() {
```

Comment:
javadocs :)

Reply:
Yeah, the Javadoc is on `toCollection()` now.

Reply:
Add at least a "As defined by ..."